server/utils/hooks.py

```
@@ -0,0 +1,98 @@
+
+    # Try custom module first
+    if _hooks_module and hasattr(_hooks_module, name):
+        fn = getattr(_hooks_module, name)
+    else:
+        fn = _HOOK_REGISTRY[name]
+
+    # Wrap in safety net so a broken hook never crashes the caller
+    def _safe_hook(*args, **kwargs):
+        try:
+            return fn(*args, **kwargs)
+        except Exception as e:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)